src/rust/rust.pro

```diff
@@ -34,9 +50,25 @@ condure_bin.target = $$bin_dir/condure
 condure_bin.depends = condure_build
 condure_bin.commands = mkdir -p $$bin_dir && cp -a $$target_dir/condure $$bin_dir/condure
 
-pushpin_bin.target = $$root_dir/pushpin
-pushpin_bin.depends = pushpin_build
-pushpin_bin.commands = cp -a $$target_dir/pushpin $$root_dir/pushpin
+m2adapter_bin.target = $$bin_dir/m2adapter
+m2adapter_bin.depends = m2adapter_build
+m2adapter_bin.commands = mkdir -p $$bin_dir && cp -a $$target_dir/m2adapter $$bin_dir/m2adapter
```

Comment:
do we need `mkdir -p $$bin_dir` cmd to be repeated for these bins?

Reply:
Sort of. The bin dir isn't part of the repo and is created on demand. Since it is possible for make to run individual targets or to run these targets in different orders, they all try to make the bin dir.

Reply:
in that case do we need to add the `mkdir` to runner_legacy_bin.commands & runner_bin.commands ?

Reply:
ah, the runner(s) go in the root dir not the bin dir, so they don't need it.